apps/user-portal/src/components/account-recovery-email.tsx

```diff
@@ -0,0 +1,271 @@
+                        <List.Content floated="right">
+                            {email || email != ""
+                                ? <Icon
+                                    link
+                                    onClick={handleEdit}
+                                    className="list-icon"
+                                    size="small"
+                                    color="grey"
+                                    name="pencil alternate"
+                                />
+                                : <Icon
+                                    link
+                                    onClick={handleEdit}
+                                    className="list-icon"
+                                    size="small"
+                                    color="grey"
+                                    name="plus"
+                                />}
+                        </List.Content>
+                    </Grid.Column>
+                </Grid.Row>
+            </Grid>
+            );
+        }
+        return (
+            <EditSection>
+                <Grid>
+                    <Grid.Row>
+                        <Grid.Column>
+                            <List>
+                                <List.Item>
+                                    <List.Content>
+                                        <Form>
+                                            <Form.Field width={9}>
+                                                <label>{t("views:accountRecovery.emailRecovery.emailAddress")}</label>
+                                                <input onChange={handleFieldChange} value={editedEmail} />
+                                                <br /><br />
+                                                <p style={{ fontSize: "12px" }}>
+
+                                                </p>
+                                            </Form.Field>
+                                        </Form>
+                                    </List.Content>
+                                </List.Item>
+                                <Divider hidden />
+                                <List.Item>
+                                    <Button primary onClick={handleUpdate} size="small">
+                                        {t("common:update")}
+                                    </Button>
+                                    <Button className="link-button" onClick={handleCancel} size="small">
+                                        {t("common:cancel")}
+                                    </Button>
+                                </List.Item>
+                            </List>
+                        </Grid.Column>
+                    </Grid.Row>
+                </Grid>
+            </EditSection>
+        );
+    };
+
+    useEffect(() => {
+        getProfileInfo()
+            .then((response) => {
+                setEmailAddr(response);
+            });
+
```

Comment:
Handle the `return` here. Else there will be continuous call to the method. As `useEffect` hook uses on both `componentDidMount` & `componentDidUpdate`